src/Pmi/Controller/BiobankController.php

```diff
@@ -33,15 +33,28 @@ public function homeAction(Application $app)
 
     public function participantsAction(Application $app, Request $request)
     {
+        $bioBankIdPrefix = $app->getConfig('biobank_id_prefix');
         $idForm = $app['form.factory']->createNamedBuilder('id', Type\FormType::class)
             ->add('biobankId', Type\TextType::class, [
                 'label' => 'Biobank ID',
                 'constraints' => [
                     new Constraints\NotBlank(),
-                    new Constraints\Type('string')
+                    new Constraints\Type('string'),
+                    new Constraints\Callback(function($bioBankId, $context) use ($bioBankIdPrefix) {
```

Comment:
Instead of a complex callback constraint, I think it would be more straightforward to use a [Regex](https://symfony.com/doc/current/reference/constraints/Regex.html) constraint. You can see an example usage of this in the `OrderController` for validating kit ids. The only downside is that you will need to build the contraints array first and dynamically add the regex constraint if the prefix is configured, but I think the benefits of using a simple constraint are worth it.